force-app/main/default/lwc/cryptoInvestor/cryptoInvestor.js

```diff
@@ -0,0 +1,630 @@
+    
+        if(emptyDataTable){
+            if (emptyDataTable) {
+                const hasData = this.transactionsHistoryData && this.transactionsHistoryData.length > 0;
+                emptyDataTable.classList.toggle('slds-hide', hasData);
+            }
+        }
+    }
+
+    displayRecordPerPageToHistoryTab(page) {
+
+        page = Math.max(1, Math.min(page, this.numberOfPagesToHistoryTab));
+
+        this.pageNumberToHistoryTab = page;
+    
+        this.startingPageIndexToHistoryTab = parseFloat(page - 1) * PAGE_SIZE;
+        this.endingPageIndexToHistoryTab = Math.min(page * PAGE_SIZE, this.transactionsHistoryData.length);
+
+        this.dataTableToHistoryTab = this.transactionsHistoryData.slice(this.startingPageIndexToHistoryTab, this.endingPageIndexToHistoryTab);
+    
+        this.isPaginatorToHistoryTransactions = this.transactionsHistoryData.length >= (parseFloat(PAGE_SIZE+1));
+        
+    }
+
+
+    nextHandlerToHistoryTab(event){
+        if (this.pageNumberToHistoryTab < this.numberOfPagesToHistoryTab) {
+            this.pageNumberToHistoryTab += 1;
+            this.displayRecordPerPageToHistoryTab(this.pageNumberToHistoryTab);
+        }
+    }
+
+    prevHandlerToHistoryTab(event) {
+        if (this.pageNumberToHistoryTab > 1) {
+            this.pageNumberToHistoryTab -= 1;
+            this.displayRecordPerPageToHistoryTab(this.pageNumberToHistoryTab);
+        }
+    }
+
+
+    async handleChangeTime(event) {
+        this.keyTransactionTimeOption = event.target.value;
+        this.isValueChangeTime(this.keyTransactionTimeOption);
+    }
+
+    isValueChangeTime(keyTransactionTimeOption) {
+        if (keyTransactionTimeOption === 'all_time') {
+            this.keyTransactionTimeOption = '';
+            this.filterTransactionData();
+        } else if (keyTransactionTimeOption === 'setting') {
+            this.openDateRangeModal();
+        } else {
+            this.setDate(keyTransactionTimeOption);
+            this.filterTransactionData();
+        }
+    }
+
+    async openDateRangeModal() {
+        const customDateRangeTransactions = await SettingTheTimeRangeModal.open({});
+        if (customDateRangeTransactions) {
+            this.startDate = new Date(customDateRangeTransactions.startDate);
+            this.endDate = new Date(customDateRangeTransactions.endDate);
+            this.filterTransactionData();
+        }
+    }
+
+    setDate(keyTransactionTimeOption) {
+        const daysMap = {
+            '7days': 7,
+            '30days': 30,
+            '90days': 90,
+        };
+        const amountOfDays = daysMap[keyTransactionTimeOption];
+        if(!amountOfDays) {
+            return this.showToast('Error', `Date not set`, 'error');;
+        }
+        
+        this.startDate = new Date(); 
+        this.endDate = new Date(this.startDate);
+        this.endDate.setDate(this.startDate.getDate() - amountOfDays);
```

Comment:
Check if there are some functions like `addDays` in Apex

Reply:
Ok!I checked! And I don't understand why he is here for me?